sklearn/metrics/_classification.py

```diff
@@ -578,77 +588,102 @@ def multilabel_confusion_matrix(
         tp = y_true == y_pred
         tp_bins = y_true[tp]
         if sample_weight is not None:
-            tp_bins_weights = np.asarray(sample_weight)[tp]
+            tp_bins_weights = sample_weight[tp]
         else:
             tp_bins_weights = None
 
-        if len(tp_bins):
-            tp_sum = np.bincount(
-                tp_bins, weights=tp_bins_weights, minlength=len(labels)
+        if tp_bins.shape[0]:
+            tp_sum = _bincount(
+                tp_bins, xp, weights=tp_bins_weights, minlength=labels.shape[0]
```

Comment:
For consistency, I prefer `xp` to be the last `kwarg` like the other funcs we have.